api/observation_test.go

```diff
@@ -0,0 +1,662 @@
+				if count == 0 {
+					count++
+					return "v4_2,data_marking,confidence_interval,time,time,geography_code,geography,aggregate_code,aggregate", nil
+				} else if count == 1 {
+					count++
+					return "146.3,p,2,Month,Aug-16,K02000001,,cpi1dim1G10100,01.1 Food", nil
+				}
+				return "", io.EOF
+			},
+			CloseFunc: func() error {
+				return nil
+			},
+		}
+
+		mockedObservationStore := &mocks.ObservationStoreMock{
+			GetCSVRowsFunc: func(*observation.Filter, *int) (observation.CSVRowReader, error) {
+				return mockRowReader, nil
+			},
+		}
+
+		api := GetAPIWithMockedDatastore(mockedDataStore, &mocks.DownloadsGeneratorMock{}, mockedObservationStore)
+		api.router.ServeHTTP(w, r)
+		So(w.Code, ShouldEqual, http.StatusOK)
+		So(w.Body.String(), ShouldEqual, expectedDocWithSingleObservation)
+
+		So(len(mockedDataStore.GetDatasetCalls()), ShouldEqual, 1)
+		So(len(mockedDataStore.CheckEditionExistsCalls()), ShouldEqual, 1)
+		So(len(mockedDataStore.GetVersionCalls()), ShouldEqual, 1)
+		So(len(mockedObservationStore.GetCSVRowsCalls()), ShouldEqual, 1)
+		So(len(mockRowReader.ReadCalls()), ShouldEqual, 3)
+	})
+
+	Convey("A successful request to get multiple observations for a version of a dataset returns 200 OK response", t, func() {
+		r := httptest.NewRequest("GET", "http://localhost:8080/datasets/cpih012/editions/2017/versions/1/observations?time=16-Aug&aggregate=*&geography=K02000001", nil)
+		w := httptest.NewRecorder()
+
+		dimensions := []models.CodeList{
+			models.CodeList{
+				Name: "aggregate",
+				HRef: "http://localhost:8081/code-lists/cpih1dim1aggid",
+			},
+			models.CodeList{
+				Name: "geography",
+				HRef: "http://localhost:8081/code-lists/uk-only",
+			},
+			models.CodeList{
+				Name: "time",
+				HRef: "http://localhost:8081/code-lists/time",
+			},
+		}
+		usagesNotes := &[]models.UsageNote{models.UsageNote{Title: "data_marking", Note: "this marks the obsevation with a special character"}}
+
+		mockedDataStore := &storetest.StorerMock{
+			GetDatasetFunc: func(string) (*models.DatasetUpdate, error) {
+				return &models.DatasetUpdate{Current: &models.Dataset{State: models.PublishedState}}, nil
+			},
+			CheckEditionExistsFunc: func(datasetID, editionID, state string) error {
+				return nil
+			},
+			GetVersionFunc: func(string, string, string, string) (*models.Version, error) {
+				return &models.Version{
+					Dimensions: dimensions,
+					Headers:    []string{"v4_2", "data_marking", "confidence_interval", "aggregate_code", "aggregate", "geography_code", "geography", "time", "time"},
+					Links: &models.VersionLinks{
+						Version: &models.LinkObject{
+							HRef: "http://localhost:8080/datasets/cpih012/editions/2017/versions/1",
+							ID:   "1",
+						},
+					},
+					State:      models.PublishedState,
+					UsageNotes: usagesNotes,
+				}, nil
+			},
+		}
+
+		count := 0
+		mockRowReader := &mocks.CSVRowReaderMock{
+			ReadFunc: func() (string, error) {
+				if count == 0 {
+					count++
```

Comment:
`count++` could be put before the `if` statement to avoid repetition? (and bump the numbers in the conditions)

Reply:
ignored?